waspc/src/ExternalCode.hs

```diff
@@ -4,20 +4,23 @@ module ExternalCode
     , fileAbsPath
     , fileText
     , readFiles
+    , SourceExternalCodeDir
     ) where
 
 import qualified Data.Text.Lazy as TextL
 import qualified Data.Text.Lazy.IO as TextL.IO
 import Data.Text (Text)
-import qualified Path
-import qualified Path.Aliases as Path
 
 import qualified Util.IO
+import StrongPath (Path, Abs, Rel, Dir, (</>))
+import qualified StrongPath as SP
 
+-- | External code directory in Wasp source, from which external code files are read.
+data SourceExternalCodeDir
 
 data File = File
-    { _pathInExtCodeDir :: !Path.RelFile  -- ^ Path relative to external code directory.
-    , _extCodeDirPath :: !Path.AbsDir  -- ^ Absolute path of external code directory.
+    { _pathInExtCodeDir :: !(Path (Rel SourceExternalCodeDir) SP.File)
```

Comment:
How cool is this!? No more comments because it is described by the type system :)!

Reply:
True, I didn't want to refactor that part, plus you can see from the type to which external code dir it refers to so it is not that important.